src/web/handlers/PinchGestureHandler.ts

```diff
@@ -0,0 +1,150 @@
+import { State } from '../../State';
+import { DEFAULT_TOUCH_SLOP } from '../constants';
+import { EventTypes, GHEvent } from '../tools/EventManager';
+import GestureHandler from './GestureHandler';
+import ScaleGestureDetector, {
+  ScaleGestureListener,
+} from '../detectors/ScaleGestureDetector';
+
+export default class PinchGestureHandler extends GestureHandler {
+  private scale = 1;
+  private velocity = 0;
+
+  private startingSpan = 0;
+  private spanSlop = 0;
+
+  private scaleDetectorListener: ScaleGestureListener = {
+    onScaleBegin: (detector: ScaleGestureDetector): boolean => {
+      this.startingSpan = detector.getCurrentSpan();
+      return true;
+    },
+    onScale: (detector: ScaleGestureDetector, event: GHEvent): boolean => {
+      const prevScaleFactor: number = this.scale;
+      this.scale *= detector.getScaleFactor(
+        this.tracker.getTrackedPointersNumber()
+      );
+
+      const delta = detector.getTimeDelta();
+      if (delta > 0) {
+        this.velocity = (this.scale - prevScaleFactor) / delta;
+      }
+
+      if (
+        Math.abs(this.startingSpan - detector.getCurrentSpan()) >=
+          this.spanSlop &&
+        this.currentState === State.BEGAN
+      ) {
+        this.activate(event);
+      }
+      return true;
+    },
+    onScaleEnd: (_detector: ScaleGestureDetector, event: GHEvent): void => {
+      this.end(event);
+      this.reset();
+    },
+  };
+
+  private scaleGestureDetector: ScaleGestureDetector = new ScaleGestureDetector(
+    this.scaleDetectorListener
+  );
+
+  public init(ref: number, propsRef: React.RefObject<unknown>) {
+    super.init(ref, propsRef);
+
+    this.setShouldCancelWhenOutside(false);
+  }
+
+  protected transformNativeEvent(_event: GHEvent) {
+    return {
+      focalX: this.scaleGestureDetector.getFocusX(),
+      focalY: this.scaleGestureDetector.getFocusY(),
+      velocity: this.velocity,
+      scale: this.scale,
+    };
+  }
+
+  get name(): string {
+    return 'pinch';
+  }
+
+  protected onDownAction(event: GHEvent): void {
+    super.onDownAction(event);
+
+    this.tracker.addToTracker(event);
+
+    if (this.tracker.getTrackedPointersNumber() < 2) return;
+
+    if (this.tracker.getTrackedPointersNumber() > 1) {
+      event.eventType = EventTypes.POINTER_DOWN;
+      this.checkUndetermined(event);
+      this.scaleGestureDetector.onTouchEvent(event, this.tracker);
+    }
+  }
+  protected onUpAction(event: GHEvent): void {
+    if (this.tracker.getTrackedPointersNumber() > 1) {
+      event.eventType = EventTypes.POINTER_UP;
```

Comment:
To be honest, I don't think moving it to pointer tracker changes that much. It should be better to do this in EventManager, but this would require much more changes. For now I'll move it to detectors, since they have access to tracker and only they are actually using this flag